src/main/java/org/folio/rest/impl/LoansAPI.java

```diff
@@ -180,6 +186,54 @@ public void postLoanStorageLoansAnonymizeByUserId(
     });
   }
 
+  @Override
+  public void postLoanStorageAnonymizeLoans(AnonymizeLoansRequest request,
+                                            Map<String, String> okapiHeaders,
+                                            Handler<AsyncResult<Response>> responseHandler,
+                                            Context vertxContext) {
+
+    AnonymizeLoansResponse response = new AnonymizeLoansResponse();
+
+    List<String> loanIds = request.getLoanIds();
+    final Set<String> invalidUuids = loanIds.stream()
+      .filter(id -> !UUIDValidation.isValidUUID(id))
```

Comment:
Why does it matter if the input are UUIDs or not?

Reply:
For optimization purposes: do not execute SQL when we know in advance uuids are invalid. They are used in the where...in clause in SQL query and  If we provide something that is not UUID, the database will throw error.

Reply:
This has been reworked.